### DISCUSSION TITLE
Multiple firmware versions

### DISCUSSION THREAD

**Discussion by G4GNX** [Jyers/Marlin#509] [Ideas]
I wonder if it would be possible to have only 2 different versions of firmware, with menus to choose required facilities and configurations. Maybe it would be too much work, but it would simplify distribution and save the issue of so many binaries every time a change is made.
2 versions would seem logical to cope with 4.2.2 and 4.2.7 boards, but I guess it could be possible to have just one version with either user choice or maybe it's possible for the firmware itself to detect which board it's in?

**Comment by smooreace** [comment]
You realize it probably takes a maximum of maybe 2 minutes to compile 40
versions, right? The only thing easier would be to not compile any and let
everyone think for themselves and compile their own and provide no support.

Thats My Vote!

**Comment by smooreace** [comment]
First, all of those config changes are already made except a very slight
few that might crop up as the firmware ages. And all of those changes are
done through simple automation.... edit one file and done
Second, all compiling can also be done with very simple automation... click
and go drink a beer. Being pushed to GitHub is even automated.

But that ok... you don't get it. Fortunately, many others do! But lets get
to the real issue... you having issues figuring out what firmware is right
for you. That too is even ok!

**Comment by Jyers** [comment]
Okay guys, chill. No need to be hostile. To better answer the question, I have all of the configuration being built by a GitHub workflow. Anytime I publish a new release, it uses a few configuration matrices to create the many different versions. Then it compiles them and uploads them to the release assets automatically. The other reason I can't make one version that accommodates every config is that many configuration changes can't be changed after compiling. BL Touch, probe grid, maxtemp, etc... All of these are compile time only options, so there is no way for them to be modifiable from the menu.

Thanks a ton for trying to reduce my workload and it's a really cool idea, just not practical with the way marlin handles a lot of settings.

**Reply by G4GNX** [reply]
That's fine Jake, I was hoping you'd respond, as you would be the one involved.
I suspected that it might be either too difficult or impossible, but thought I'd chuck it in anyway.
Ho-hum, back to the drawing board..............

Thanks for the excellent work so far. Not only is it fun to explore, it's great to use.
